### ISSUE TITLE
Crash on load of corrupted project

### ISSUE THREAD

**Issue by Tisten** [OurMachinery/themachinery-public#1185]
- **Version:** Git Master Branch
- **OS:** Windows

The-machinery crashed when trying to load the clouds project posted about in discord here: https://discord.com/channels/690290013585408136/691973584612229191/953721773474603018
Saved a copy here: [clouds.zip](https://github.com/OurMachinery/themachinery-public/files/8284157/clouds.zip)

The project was corrupted in some way, and a new version was uploaded afterwards. But I would expect the editor to discover the problem and abort loading instead of crashing. I would like a pop-up message saying that the project couldn't load, and as good info as possible about what data was bad.

**Comment by niklas-ourmachinery** [comment]
Agreed.

I don't think we can come up with a generic mechanism that can detect all kinds of project corruption. The only thing I can think of that can do that is to do a hash/checksum of all the files in the project (or the whole single file in case of a database project)... but we do on-demand/lazy loading of data, so we don't want to add a step that loads everything.

Instead, I think we need to address this in a one-by-one whack-a-mole fashion. I.e. whenever we encounter a crash from loading a corrupt project, we fix it so that that particular corruption causes an error message instead of a crash.

And in addition, we should do some fuzzy testing -- randomly flip bits in the project data and make sure that we don't crash.

This might not get rid of *all* corruption crashes, but it should hopefully cover most of them.

**Comment by leonardo-ourmachinery** [comment]
I think the same thinking process applies to "data sanitization": me and @Tisten encountered a bug where two different entity assets had the same uuid in the "components" property. This of course should never happen and was probably due to a bug we had in the past.

But I guess we'll have to tackle those data corruption bugs one by one as we discover them, and add them to our "data sanification tool" when that happens, if possible.

**Comment by leonardo-ourmachinery** [comment]
I looked into this, and pushed those two fixes: https://github.com/OurMachinery/themachinery/commit/155d6803c0cab0212055563adf65507dede6abcf
https://github.com/OurMachinery/themachinery/commit/f25cb34904cbedadc3ec120fe6f9f546da37d361

Now the "Corrupted project" loading experience is what I would expect, with specific errors printed in the console ("Bad Asset format") and a big modal saying that "Project foo cannot be loaded".

Not sure what's up with that "Bad asset format" error, I assume that at some point in the past we broke backward compatibility for some reason. @niklas-ourmachinery 

@Tisten If you're happy with this I think we can close the issue and address corruption crashes as they come.
